main.go

```diff
@@ -418,11 +442,37 @@ func main() { //nolint:funlen,maintidx
 	}
 }
 
-func createSecretCacheConfig(platform cluster.Platform) map[string]cache.Config {
+func createSecretCacheConfig(ctx context.Context, cli client.Client, upgrade bool, platform cluster.Platform) map[string]cache.Config {
 	namespaceConfigs := map[string]cache.Config{
-		"istio-system":      {}, // for both knative-serving-cert and default-modelregistry-cert,as an easy workarond, to watch all in this namespace for now
+		"istio-system":      {}, // for both knative-serving-cert and default-modelregistry-cert, as an easy workarond, to watch both in this namespace
 		"openshift-ingress": {},
 	}
+	// upgrade cache
+	if upgrade {
```

Comment:
not sure if this `upgrade` flag is really meaningful because my understanding is that this condition would be triggered every time the operator starts once it is installed, so it wont fall into the `opendatahub.io/application-namespace` case.